packages/design-system/src/scalars/components/number-field/number-field.tsx

```diff
@@ -26,7 +27,7 @@ export const NumberFieldRaw = forwardRef<HTMLInputElement, NumberFieldProps>(
       description,
       value,
       defaultValue,
-      onChange,
+      onChange = () => {},
```

Comment:
`onChange` should be optional without a default value, Adding an "empty function" is going to still be a function and could change how the validation is done making it less efficient

Reply:
done